SourcePointMetaApp/SourcePointMetaAppTests/ConsentViewControllerTests/ConsentViewControllerTest.swift

```diff
@@ -0,0 +1,295 @@
+        }
+    }
+    
+    /// this method is used to test whether webview is loaded or not
+    func testLoadView() {
+        consentViewController?.loadView()
+        if consentViewController?.webView != nil {
+            XCTAssert(true, "Webview initialized successfully")
+        } else {
+            XCTAssert(false, "Unable to initialize webview")
+        }
+    }
+    
+    /// this method is used to test whether message url is created or not successfully
+    func testGetMessageUrlWithAuthId() {
+        let messageUrl = consentViewController!.getMessageUrl(authId: authId)
+        if messageUrl != nil {
+            XCTAssert(true, "Message URL created successfully")
+        } else {
+            XCTAssert(false, "Unable to create Message URL")
+        }
+    }
+    
+    /// this method is used to test whether message url is created or not successfully
+    func testGetMessageUrlWithoutAuthId() {
+        let messageUrl = consentViewController!.getMessageUrl(authId: nil)
+        if messageUrl != nil {
+            XCTAssert(true, "Message URL created successfully")
+        } else {
+            XCTAssert(false, "Unable to create Message URL")
+        }
+    }
+    
+    /// this method is used to test whether message is loaded or not successfully
+    func testLoadMessageWithAuthId() {
+        consentViewController?.loadMessage(forAuthId: authId)
+        if consentViewController?.webView != nil {
+            XCTAssert(true, "Webview initialized successfully")
+        } else {
+            XCTAssert(false, "Unable to initialize webview")
+        }
+    }
+    
+    /// this method is used to test whether message is loaded or not successfully
+    func testLoadMessageWithoutAuthId() {
+        consentViewController?.loadMessage()
+        if consentViewController?.webView != nil {
+            XCTAssert(true, "Webview initialized successfully")
+        } else {
+            XCTAssert(false, "Unable to initialize webview")
+        }
+    }
+    
+    /// this method is used to test whether GDPR status method is working or not
+    func testSetSubjectToGDPR() {
+        var finished = false
+        consentViewController?.setSubjectToGDPR()
+        if UserDefaults.standard.string(forKey: ConsentViewController.IAB_CONSENT_SUBJECT_TO_GDPR) != nil {
+            if UserDefaults.standard.string(forKey: ConsentViewController.IAB_CONSENT_SUBJECT_TO_GDPR) == "0" || UserDefaults.standard.string(forKey: ConsentViewController.IAB_CONSENT_SUBJECT_TO_GDPR) == "1"  {
+                XCTAssert(true, "succeed to get GDPR status from endpoint")
+            }else {
+                XCTAssert(false, "failed to get GDPR status from endpoint")
+            }
+            finished = true
+        }
+        while !finished {
+            RunLoop.current.run(mode: .default, before: Date.distantFuture)
+        }
+    }
+    
+    /// this method is used to test whether getIABVendorConsents method is able to get vendors or not from server
```

Comment:
* the comment is incorrect. All iAB-related consents are retrieved from the `UserDefaults` and not from the server.
* we should assert upon the real return value rather than `!= nil` only. This is valid for all functions below.

Reply:
Updated the comment.

Reply:
Asserts are used on real values